app/assets/javascripts/provider/admin/apiconfig/services/integration.js.coffee

```diff
@@ -208,17 +208,21 @@ $(document).on 'initialize', '#proxy', ->
     false
 
   proxy_credentials_location_value = $('input[name="proxy[credentials_location]"]:checked').val()
+  credentials_forms = document.querySelectorAll('li[id^="proxy_auth_"]')
+  credentials_forms.forEach((el) -> el.classList.toggle('hidden', proxy_credentials_location_value is 'authorization'))
 
   replace_hyphens_or_underscores = (elt) ->
     return unless elt
     elt.value = switch proxy_credentials_location_value
       when 'headers' then elt.value.replace(/_/g,'-')
       when 'query' then elt.value.replace(/-/g,'_')
+      else elt.value
 
   $(document).on 'change', 'input[name="proxy[credentials_location]"]', (event) ->
     proxy_credentials_location_value = this.value
     replace_hyphens_or_underscores(document.getElementById('proxy_auth_user_key'))
     toggle_form_changed_ui()
+    credentials_forms.forEach((el) -> el.classList.toggle('hidden'))
```

Comment:
Isn't this toggling it every time? Even when selecting "http headers" option?

Reply:
Indeed, good catch!